managers/docker-manager/DockerManager.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const Promise = require('bluebird');
+const eventmesh = require('../../data-access-layer/eventmesh');
+const logger = require('../../common/logger');
+const CONST = require('../../common/constants');
+const utils = require('../../common/utils');
+const BaseManager = require('../BaseManager');
+const DockerService = require('./DockerService');
+const errors = require('../../common/errors');
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+
+class DockerManager extends BaseManager {
+
+  init() {
+    const validStateList = [CONST.APISERVER.RESOURCE_STATE.IN_QUEUE, CONST.APISERVER.RESOURCE_STATE.UPDATE, CONST.APISERVER.RESOURCE_STATE.DELETE];
+    return this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT, CONST.APISERVER.RESOURCE_TYPES.DOCKER)
+      .then(() => this.registerWatcher(CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT, CONST.APISERVER.RESOURCE_TYPES.DOCKER, validStateList));
+  }
+
+  processRequest(changeObjectBody) {
+    return Promise.try(() => {
+        if (changeObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.IN_QUEUE) {
```

Comment:
Better to use switch over if/else ladder for better readability. 
Also add default with log.warn saying not supported state.

Reply:
Added a TODO